### PR TITLE
Behavior editor

### PR DESCRIPTION
**Pull request by balderdash**
This adds a "Behavior"-typed toolbox category that includes all the behaviors defined in your program, complete with "Create a Behavior" button:
image

You also get a modal behavior editor when you create/edit a behavior. For now you just get one param named `this sprite`:
image

### CODE REVIEW

**Comment by balderdash** [comment]
[See inside behaviors as functions (Support for user-editable behaviors)](https://trello.com/c/uUs2xCMj/81-see-inside-behaviors-as-functions-support-for-user-editable-behaviors)

File: apps/src/gamelab/GameLabJr.interpreted.js [hunk 1]
```diff
@@ -67,12 +67,16 @@ function removeBehavior(sprite, behavior) {
...
+  };
+}
+
 function normalizeBehavior(behavior) {
   if (typeof behavior === 'function')  {
-    behavior = {
-      func: behavior,
-      extraArgs: [],
-    };
+    behavior = createBehavior(behavior, []);
```
Comment (Hamms) [inline apps/src/gamelab/GameLabJr.interpreted.js]: I don't understand the purpose of this change

File: apps/src/gamelab/blocks.js [hunk 2]
```diff
@@ -551,14 +593,143 @@ export default {
...
+    Blockly.Variables.getters[Blockly.BlockValueType.BEHAVIOR] = 'gamelab_behavior_get';
+    Blockly.Flyout.configure(Blockly.BlockValueType.BEHAVIOR, {
+      initialize(flyout, cursor) {
+        if (Blockly.functionEditor && !Blockly.functionEditor.isOpen()) {
+          flyout.addButtonToFlyout_(cursor, 'Create a Behavior',
+            behaviorEditor.openWithNewFunction.bind(behaviorEditor));
+        }
+      },
+      addDefaultVar: false,
+    });
```
Comment (Hamms) [inline apps/src/gamelab/blocks.js:678]: I don't love how much manual work is required here to get this to work, particularly given how much of it is just copying stuff from `procedures_defnoreturn`. Can some of this be pushed into Blockly and generalized? (possibly in a separate PR)

File: apps/src/gamelab/GameLabJr.interpreted.js (hunk 1 above)
Comment (balderdash) [reply apps/src/gamelab/GameLabJr.interpreted.js]: Ah, I meant to use this function in the behavior getter block, I haven't added support for parameterized behaviors to these blocks yet, so this isn't needed.

Reverted for now.

File: apps/src/gamelab/blocks.js (hunk 2 above)
Comment (balderdash) [reply apps/src/gamelab/blocks.js:678]: I added a helper in https://github.com/code-dot-org/blockly/pull/107 and used it in https://github.com/code-dot-org/code-dot-org/pull/22307/commits/1a2037d411184c3ebc4ea7d5438426c5fc4e8aa5, cutting out quite a lot of the extra junk here.

**Pull request merged**